packages/fxa-auth-db-mysql/lib/db/mem.js

```diff
@@ -1497,6 +1498,55 @@ module.exports = function (log, error) {
       })
   }
 
+  Memory.prototype.createAccountSubscription = function (uid, subscriptionId, productName, createdAt) {
+    uid = uid.toString('hex')
+    return getAccountByUid(uid).then(() => {
+      const key = [uid, subscriptionId, productName].join('|')
+      if (accountSubscriptions[key]) {
+        return P.reject(error.duplicate())
+      }
+      const existingSubs = Object
```

Comment:
What does this search protect against in addition to checking the key directly in line 1505?

Reply:
Trying to mimic the unique indexes in the DB table. The key at 1505 is unique for the combination of uid + subscriptionId + productName - e.g. 1505 alone would allow two different UIDs to have the same subscriptionId. This one enforces subscriptionId alone as unique across the table.